src/main/java/org/myrobotlab/service/abstracts/AbstractSpeechSynthesis.java

```diff
@@ -901,7 +906,24 @@ public void stop() {
   public String setAudioEffects(String audioEffects) {
     return audioEffects;
   }
-  
+
+  // TODO check also internet for cloud services, not only api
+  @Override
+  public boolean isReady() {
+    String[] kn = getKeyNames();
```

Comment:
This logic should not be here.  when the service is started, is when it should decided that the service is ready.  This should only call the default behavior of returning this.ready  (from the base class.)

Reply:
Right, no need to override - you can check in the constructor and
isReady = true/false depending on logic there

Reply:
Inside the constructor is not enough, There is events on startService() that can take long time for service start. 
Also we need to "recheck" things sometime. by example if user setKeys() after service start. We need to set ready=true. 
setKeys it's not necessarily the only one check, we can have many more